data/scrapper/erplogin/erpSession.py

```diff
@@ -0,0 +1,141 @@
+import requests
+import getpass
+
+ERP_HOME_URL = 'https://erp.iitkgp.ac.in/IIT_ERP3/' #to get the session cookie
+SECURITY_QUESTION_URL = 'https://erp.iitkgp.ac.in/SSOAdministration/getSecurityQues.htm'
+LOGIN_URL = 'https://erp.iitkgp.ac.in/SSOAdministration/auth.htm'
+GET_ACAD_TOKEN_URL = 'https://erp.iitkgp.ac.in/Acad/central_breadth_tt.jsp?action=second'
+
+
+class Session:
+    """
+    An erp session
+
+    Logs into erpr after prompting the user to enter details
+
+    
+    Class Attributes:
+    >>>self.sessionToken
+    >>>self.ssoToken
+    >>>self.academicToken
+
+    Basic Usage::
+
+    >>>from erplogin import erpSession
+    >>>s = erpSession.Session()
+    
+    >>>academicToken = s.academicToken
+    """
+
+    headers = {
+        'timeout': '20',
+        'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/51.0.2704.79 Chrome/51.0.2704.79 Safari/537.36',
+    }
+
+    def __init__(self):
+        self.sess = requests.Session()
+        self.sessionToken = Session.__generate_session_cookie(self)
+        self.ssoToken = Session.__generate_sso_token(self)
+        self.academicToken = Session.__generate_acad_cookie(self)
+
+    def __generate_session_cookie (self):
+        """
+        Requests for the session id by requesting for erp home page.
+
+        Arguments:
+        > self
+
+        Returns 
+        > sessionToken
+        """
+        print("Getting session cookies")
+        response_erp = self.sess.get(ERP_HOME_URL)
+        session_token = response_erp.cookies['JSESSIONID']
+
+        return (session_token)
+
+
+    def __generate_sso_token(self):
+        """
+        Prompts the user to enter the details and sign into erp with the entered details
+
+        Arguments:
+        > self
+
+        Returns 
+        > ssoToken
+        """
+        while (1):
+            while (1):
+                roll_no = input("Enter your roll number : ")
```

Comment:
Modules are not supposed to be designed to be user interactive i.e. should not use `input` or `print`.
Pass the required data(roll no, password and security answer) into the function as parameters. You can write a generic function called `LoginERP` here which does the I/O. This function can then be used by the scrapping scripts for login.

Reply:
The security answer can be one of the three questions, add to get it, one needs to obtain the sessionToken.

Reply:
Ping @kshitij10496 .

Reply:
@kshitij10496 Can you explain how this will work?

Reply:
Here is the UI I would like to use in the scripts.

```python
s = ERPSession(roll_no, password)
question = s.get_security_question()
# User feeds the answer 
s.Login(answer)
```

Thus, `ERPSession` should have fields `roll_no`, `password` and `security_question` and `security_answer` in addition to the tokens.

Reply:
Okay :smile:

Reply:
@kshitij10496 Won't having roll_no or password as attributes to the class be problematic

Reply:
As of now I am writing code that stores the password as attribute

Reply:
Good question. 🤔 
Can you elaborate on why do you think this would be an issue?

Reply:
The only reason is that password will be accessible by the user at a later point of time as long as its logged in the same erpsession till the time, I reassign it to empty string after logging in.

Reply:
Ping @kshitij10496

Reply:
You can make the "password" private by naming it as `_password`.
Moreover, I do not think there would be any harm in allowing the user to access his own password anyway.